src/bp/core/services/telemetry/configs.ts

```diff
@@ -0,0 +1,170 @@
+      config,
+      (res, value, key) => {
+        if (SECRET_KEYS.find(x => key.toLowerCase().includes(x))) {
+          res[key] = _.isEqual(config[key], defaultConfig[key]) ? DEFAULT : REDACTED
+        } else if (!_.isArray(value) && _.isObject(value)) {
+          res[key] = this.obfuscateSecrets(value, defaultConfig)
+        } else {
+          res[key] = value
+        }
+        return res
+      },
+      {}
+    )
+  }
+
+  private async fetchSchema(schemaName: string): Promise<any> {
+    try {
+      return await this.ghostService.root().readFileAsObject('/', schemaName)
+    } catch (error) {
+      return {}
+    }
+  }
+
+  private async getBotpressConfigs(): Promise<BotpressConfig> {
+    const botpressConfig = await this.config.getBotpressConfig()
+    const defaultConfig = await this.fetchSchema('botpress.config.schema.json')
+
+    return this.obfuscateSecrets(botpressConfig, defaultConfig)
+  }
+
+  private async getModulesConfigs(): Promise<ModuleConfigEvent[]> {
+    const bots = await this.botService.getBotsIds()
+    const modules = _.intersection(BUILTIN_MODULES, Object.keys(process.LOADED_MODULES))
+
+    return (await Promise.map(modules, this.getConfigsByModule(bots))).reduce((acc, cur) => [...acc, ...cur])
+  }
+
+  private getConfigsByModule(bots: string[]): (module: string) => Promise<ModuleConfigEvent[]> {
+    return async module => {
+      const defaultValue = await this.moduleLoader.configReader.loadFromDefaultValues(module)
+      return Promise.map(bots, this.getModuleConfigPerBot(module, defaultValue))
+    }
+  }
+
+  private getModuleConfigPerBot(module: string, defaultValue: any): (botId: any) => Promise<ModuleConfigEvent> {
+    return async botId => {
+      const runtimeValue = await this.moduleLoader.configReader.getForBot(module, botId)
+      return { botId, module, configs: this.obfuscateSecrets(runtimeValue, defaultValue) }
+    }
+  }
+
+  private async getBotsConfigs(): Promise<BotConfigEvent[]> {
+    const defaultConfig = defaultJsonBuilder(await this.fetchSchema('bot.config.schema.json'))
+    const bots = Array.from(await this.botService.getBots())
+
+    return bots.reduce((acc: any[], bot) => {
+      return [...acc, { botId: calculateHash(bot[0]), botConfigs: this.formatBotConfig(bot[1], defaultConfig) }]
+    }, [])
+  }
+
+  private formatBotConfig(botConfig: BotConfig, defaultConfig): BotConfig {
+    return {
+      ...this.obfuscateSecrets(botConfig, defaultConfig),
+      details: this.formatBotDetails(botConfig.details),
+      id: calculateHash(botConfig.id),
+      name: calculateHash(botConfig.name)
+    }
+  }
+
+  private formatBotDetails(details: BotDetails): BotDetails {
+    const detailKeys = [
+      'website',
+      'phoneNumber',
+      'termsConditions',
+      'emailAddress',
+      'avatarUrl',
+      'coverPictureUrl',
+      'privacyPolicy'
+    ]
+    return detailKeys.reduce((acc, key) => ({ ...acc, [key]: details[key] ? 'redacted' : 'default' }), {})
```

Comment:
Can you use your existing const here ?

Reply:
fixed in: 6ee921600c47c73f8f8a254bf0e83f7d735de5ea